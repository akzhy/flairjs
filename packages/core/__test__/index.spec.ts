import { test, expect, describe } from 'vitest'

import { transformCode } from '../index'
import { readdirSync, readFileSync } from 'node:fs'
import { fileURLToPath } from 'url'
import { dirname } from 'path'
import path from 'node:path'

const __filename = fileURLToPath(import.meta.url)
const __dirname = dirname(__filename)
const styleTagContent = readFileSync(path.resolve(__dirname, './snippets/style-tag.tsx'), 'utf-8')
const flairPropertyContent = readFileSync(path.resolve(__dirname, './snippets/flair-string.tsx'), 'utf-8')
const flairPropertyObjectContent = readFileSync(path.resolve(__dirname, './snippets/flair-obj.tsx'), 'utf-8')

describe('Core tests', () => {
  test('style tag is working', () => {
    const result = transformCode(styleTagContent, 'index.tsx', {
      cssOutDir: path.resolve(__dirname, './.css'),
    })
    if (!result) {
      throw new Error('transformCode returned null or undefined')
    }
    expect(result.code).toMatchSnapshot()
  })

  test('flair property string is working', () => {
    const result = transformCode(flairPropertyContent, 'index.tsx', {
      cssOutDir: path.resolve(__dirname, './.css'),
    })
    if (!result) {
      throw new Error('transformCode returned null or undefined')
    }
    expect(result.code).toMatchSnapshot()
  })

  test('flair property object is working', () => {
    const result = transformCode(flairPropertyObjectContent, 'index.tsx', {
      cssOutDir: path.resolve(__dirname, './.css'),
    })
    const cssFiles = readdirSync(path.resolve(__dirname, './.css')).filter((f) => f.endsWith('.css'))
    const cssContent = readFileSync(path.resolve(__dirname, './.css', cssFiles[0]), 'utf-8')


    expect(cssContent).toMatchSnapshot()

    if (!result) {
      throw new Error('transformCode returned null or undefined')
    }
    expect(result.code).toMatchSnapshot()
  })
})
