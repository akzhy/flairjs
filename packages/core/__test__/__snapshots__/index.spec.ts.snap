// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`flair property object is working 1`] = `
"import "@flairjs/css/cached-css/6091365a83003455.css";
import clsx from "clsx";
import { c, cn, flair } from "@flairjs/react";
import { useState } from "react";
export const TestCaseComponent = () => {
	const [case5, setCase5] = useState(false);
	const [case8, setCase8] = useState(false);
	const [case9, setCase9] = useState(false);
	const case3 = "_8Z4fiW_case-3";
	const case6_3 = "_8Z4fiW_case-6-3";
	const case6_4 = "case-6-4";
	const case9_1 = "case-9 _8Z4fiW_case-9-1";
	const case9_2 = "case-9 _8Z4fiW_case-9-2";
	const case10 = "case-10";
	const case11_1 = "case-11-1";
	const case11 = \`case-11 \${case11_1} _8Z4fiW_case-11-2\`;
	const case12_1 = "case-12-1";
	const case12 = "case-12 " + case12_1 + " _8Z4fiW_case-12-2";
	const case13 = c("_8Z4fiW_case-13 case-13-1");
	const case14 = cn("_8Z4fiW_case-14 case-14-1");
	const case15Handler = () => {
		return c("_8Z4fiW_case-15 case-15-1");
	};
	return <>
      <p className="_8Z4fiW_case-1">Case1</p>
      <p className="_8Z4fiW_case-2 _8Z4fiW_case-2-1 case-2-2">Case2</p>
      <div className={case3}>Case3</div>
      <div className={clsx("_8Z4fiW_case-4")}>Case4</div>
      <button className={clsx("_8Z4fiW_case-5", {
		"_8Z4fiW_case-5-1": true,
		"_8Z4fiW_case-5-2": case5,
		"case-5-3": true
	})} onClick={() => setCase5((p) => !p)}>
        Case5
      </button>
      <div className={clsx("_8Z4fiW_case-6", [
		"_8Z4fiW_case-6-1",
		"case-6-2",
		case6_3
	], { [case6_4]: true })}>
        Case6
      </div>
      <div className={["case-7", "case-7-1"].join(" ")}>Case7</div>
      <button className={case8 ? "_8Z4fiW_case-8 _8Z4fiW_case-8-1" : "_8Z4fiW_case-8 _8Z4fiW_case-8-2"} onClick={() => {
		setCase8((p) => !p);
	}}>
        Case8
      </button>
      <button className={case9 ? case9_1 : case9_2} onClick={() => {
		setCase9((p) => !p);
	}}>
        Case9
      </button>
      <div className={clsx({ [case10]: true })}>
        Case10
      </div>
      <div className={case11}>Case11</div>
      <div className={case12}>Case12</div>
      <div className={case13}>Case13</div>
      <div className={clsx(case14, "case-14-2")}>Case14</div>
      <div className={case15Handler()}>Case15</div>
    </>;
};
false;
"
`;

exports[`flair property string is working 1`] = `
"import "@flairjs/css/cached-css/6091365a83003455.css";
import clsx from "clsx";
import { c, cn } from "@flairjs/react";
import { useState } from "react";
export const TestCaseComponent = () => {
	const [case5, setCase5] = useState(false);
	const [case8, setCase8] = useState(false);
	const [case9, setCase9] = useState(false);
	const case3 = "_8Z4fiW_case-3";
	const case6_3 = "_8Z4fiW_case-6-3";
	const case6_4 = "case-6-4";
	const case9_1 = "case-9 _8Z4fiW_case-9-1";
	const case9_2 = "case-9 _8Z4fiW_case-9-2";
	const case10 = "case-10";
	const case11_1 = "case-11-1";
	const case11 = \`case-11 \${case11_1} _8Z4fiW_case-11-2\`;
	const case12_1 = "case-12-1";
	const case12 = "case-12 " + case12_1 + " _8Z4fiW_case-12-2";
	const case13 = c("_8Z4fiW_case-13 case-13-1");
	const case14 = cn("_8Z4fiW_case-14 case-14-1");
	const case15Handler = () => {
		return c("_8Z4fiW_case-15 case-15-1");
	};
	return <>
      <p className="_8Z4fiW_case-1">Case1</p>
      <p className="_8Z4fiW_case-2 _8Z4fiW_case-2-1 case-2-2">Case2</p>
      <div className={case3}>Case3</div>
      <div className={clsx("_8Z4fiW_case-4")}>Case4</div>
      <button className={clsx("_8Z4fiW_case-5", {
		"_8Z4fiW_case-5-1": true,
		"_8Z4fiW_case-5-2": case5,
		"case-5-3": true
	})} onClick={() => setCase5((p) => !p)}>
        Case5
      </button>
      <div className={clsx("_8Z4fiW_case-6", [
		"_8Z4fiW_case-6-1",
		"case-6-2",
		case6_3
	], { [case6_4]: true })}>
        Case6
      </div>
      <div className={["case-7", "case-7-1"].join(" ")}>Case7</div>
      <button className={case8 ? "_8Z4fiW_case-8 _8Z4fiW_case-8-1" : "_8Z4fiW_case-8 _8Z4fiW_case-8-2"} onClick={() => {
		setCase8((p) => !p);
	}}>
        Case8
      </button>
      <button className={case9 ? case9_1 : case9_2} onClick={() => {
		setCase9((p) => !p);
	}}>
        Case9
      </button>
      <div className={clsx({ [case10]: true })}>
        Case10
      </div>
      <div className={case11}>Case11</div>
      <div className={case12}>Case12</div>
      <div className={case13}>Case13</div>
      <div className={clsx(case14, "case-14-2")}>Case14</div>
      <div className={case15Handler()}>Case15</div>
    </>;
};
false;
"
`;

exports[`style Tag is working 1`] = `
"import "@flairjs/css/cached-css/6091365a83003455.css";
import clsx from "clsx";
import { Style, c, cn } from "@flairjs/react";
import { useState } from "react";
export const TestCaseComponent = () => {
	const [case5, setCase5] = useState(false);
	const [case8, setCase8] = useState(false);
	const [case9, setCase9] = useState(false);
	const case3 = "_8Z4fiW_case-3";
	const case6_3 = "_8Z4fiW_case-6-3";
	const case6_4 = "case-6-4";
	const case9_1 = "case-9 _8Z4fiW_case-9-1";
	const case9_2 = "case-9 _8Z4fiW_case-9-2";
	const case10 = "case-10";
	const case11_1 = "case-11-1";
	const case11 = \`case-11 \${case11_1} _8Z4fiW_case-11-2\`;
	const case12_1 = "case-12-1";
	const case12 = "case-12 " + case12_1 + " _8Z4fiW_case-12-2";
	const case13 = c("_8Z4fiW_case-13 case-13-1");
	const case14 = cn("_8Z4fiW_case-14 case-14-1");
	const case15Handler = () => {
		return c("_8Z4fiW_case-15 case-15-1");
	};
	return <>
      <p className="_8Z4fiW_case-1">Case1</p>
      <p className="_8Z4fiW_case-2 _8Z4fiW_case-2-1 case-2-2">Case2</p>
      <div className={case3}>Case3</div>
      <div className={clsx("_8Z4fiW_case-4")}>Case4</div>
      <button className={clsx("_8Z4fiW_case-5", {
		"_8Z4fiW_case-5-1": true,
		"_8Z4fiW_case-5-2": case5,
		"case-5-3": true
	})} onClick={() => setCase5((p) => !p)}>
        Case5
      </button>
      <div className={clsx("_8Z4fiW_case-6", [
		"_8Z4fiW_case-6-1",
		"case-6-2",
		case6_3
	], { [case6_4]: true })}>
        Case6
      </div>
      <div className={["case-7", "case-7-1"].join(" ")}>Case7</div>
      <button className={case8 ? "_8Z4fiW_case-8 _8Z4fiW_case-8-1" : "_8Z4fiW_case-8 _8Z4fiW_case-8-2"} onClick={() => {
		setCase8((p) => !p);
	}}>
        Case8
      </button>
      <button className={case9 ? case9_1 : case9_2} onClick={() => {
		setCase9((p) => !p);
	}}>
        Case9
      </button>
      <div className={clsx({ [case10]: true })}>
        Case10
      </div>
      <div className={case11}>Case11</div>
      <div className={case12}>Case12</div>
      <div className={case13}>Case13</div>
      <div className={clsx(case14, "case-14-2")}>Case14</div>
      <div className={case15Handler()}>Case15</div>
      
    </>;
};
"
`;
